AsyncDisplayKit/ASControlNode.mm

```diff
@@ -316,13 +298,22 @@ - (NSArray *)actionsForTarget:(id)target forControlEvent:(ASControlNodeEvent)con
 
   ASDN::MutexLocker l(_controlLock);
   
-  // Grab the event dispatch table for this event.
-  NSMapTable *eventDispatchTable = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
-  if (!eventDispatchTable)
+  // Grab the event target action array for this event.
+  NSMutableArray *eventTargetActionArray = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
+  if (!eventTargetActionArray) {
     return nil;
+  }
 
-  // Return the actions for this target.
-  return [eventDispatchTable objectForKey:target];
+  NSMutableArray *actions = [[NSMutableArray alloc] init];
+  
+  // Collect all actions for this target.
+  for (ASControlTargetAction *targetAction in eventTargetActionArray) {
+    if ((target == nil && targetAction.createdWithNoTarget) || (target != nil && target == targetAction.target)) {
+      [actions addObject:NSStringFromSelector(targetAction.action)];
```

Comment:
@appleguy If I understand you correctly then you're suggesting to maintain a dictionary where all the actions for a target stored upon adding a target action pair? This could definitely improve the performance of this code.